src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/tests/test_sql_commands.py

```diff
@@ -389,48 +413,98 @@ def test_sql_db_restore(self, resource_group, resource_group_location, server):
+                                   restore_service_objective),
+                     JMESPathCheck('status', 'Online')])
 
         # Restore to db into pool
-        db = self.cmd('sql db restore -g {} -s {} -n {} -t {} --dest-name {}'
-                      ' --elastic-pool {}'
-                      .format(rg, server, database_name, datetime.utcnow().isoformat(),
-                              restore_pool_database_name, elastic_pool),
-                      checks=[
-                          JMESPathCheck('resourceGroup', rg),
-                          JMESPathCheck('name', restore_pool_database_name),
-                          JMESPathCheck('elasticPoolName', elastic_pool),
-                          JMESPathCheck('status', 'Online')])
+        self.cmd('sql db restore -g {} -s {} -n {} -t {} --dest-name {}'
+                 ' --elastic-pool {}'
+                 .format(rg, server, database_name, datetime.utcnow().isoformat(),
+                         restore_pool_database_name, elastic_pool),
+                 checks=[
+                     JMESPathCheck('resourceGroup', rg),
+                     JMESPathCheck('name', restore_pool_database_name),
+                     JMESPathCheck('elasticPoolName', elastic_pool),
+                     JMESPathCheck('status', 'Online')])
+
+
+class SqlServerDbRestoreDeletedScenarioTest(ScenarioTest):
+    @ResourceGroupPreparer()
+    @SqlServerPreparer()
+    def test_sql_db_restore_deleted(self, resource_group, resource_group_location, server):
+        rg = resource_group
+        database_name = 'cliautomationdb01'
+
+        # Standalone db
+        restore_service_objective = 'S1'
+        restore_edition = 'Standard'
+        restore_database_name1 = 'cliautomationdb01restore1'
+        restore_database_name2 = 'cliautomationdb01restore2'
+
+        # Create database and wait for first backup to exist
+        _create_db_wait_for_first_backup(self, rg, server, database_name)
+
+        # Delete database
+        self.cmd('sql db delete -g {} -s {} -n {} --yes'.format(rg, server, database_name))
+
+        # Wait for deleted database to become visible. When run live, this will take around
+        # 5-10 minutes. Unforunately there's no way to speed this up. Use timeout to ensure
+        # test doesn't loop forever if there's a bug.
+        start_time = datetime.now()
+        timeout = timedelta(0, 15 * 60)  # 15 minutes timeout
+
+        while True:
+            deleted_dbs = list(self.cmd('sql db list-deleted -g {} -s {}'.format(rg, server)).get_output_in_json())
+
+            if deleted_dbs:
+                # Deleted db found, stop polling
+                break
+
+            # Deleted db not found, sleep (if running live) and then poll again.
+            if self.is_live:
+                self.assertTrue(datetime.now() < start_time + timeout, 'Deleted db not found before timeout expired.')
+                sleep(10)  # seconds
```

Comment:
we may want a timeout to break out of potential infinite loop

Reply:
The timeout is set on line 471 and tested on line 482.